protocol/communities_key_distributor.go

```diff
@@ -104,6 +104,7 @@ func (ckd *CommunitiesKeyDistributorImpl) distributeKey(community *communities.C
 }
 
 func (ckd *CommunitiesKeyDistributorImpl) sendKeyExchangeMessage(community *communities.Community, hashRatchetGroupID []byte, pubkeys []*ecdsa.PublicKey, msgType common.CommKeyExMsgType) error {
+	// NOTE: should the sender by set to the community control node?
```

Comment:
What do you mean here? In theory, only the Owner should be able to send encryption keys. As for Owner vs  control node, unless we send the installation ID, it's both the same pubkey.

Reply:
Aaah, I get it. I guess it would be more rigorous to set it to the community key while the Owner token is not minted, but maybe you can keep the code simpler by using the Owner's pubkey at all times?

Reply:
Received encryption keys aren't validated, so anyone could send them. Once validation is implemented, the sender must indeed be the community itself for non-tokenized communities.

Reply:
probably I should make that change now (not the validation), but at least send from control node, otherwise once we add it, it will not validate anymore

Reply:
+1